controllers/imagecollector/imagecollector_controller.go

```diff
@@ -0,0 +1,342 @@
+	if until > 0 {
+		log.Info("Delaying imagejob delete", "job", job.Name, "deleteAter", job.Status.DeleteAfter)
+		return ctrl.Result{RequeueAfter: until}, nil
+	}
+
+	log.Info("Deleting imagejob", "job", job.Name)
+	err := r.Delete(ctx, job)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	log.Info("end job deletion")
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) createImageJob(ctx context.Context, req ctrl.Request, imageCollector *eraserv1alpha1.ImageCollector) (ctrl.Result, error) {
+	job := &eraserv1alpha1.ImageJob{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "imagejob-",
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(imageCollector, schema.GroupVersionKind{Group: "eraser.sh", Version: "v1alpha1", Kind: "ImageCollector"}),
+			},
+		},
+		Spec: eraserv1alpha1.ImageJobSpec{
+			JobTemplate: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy: corev1.RestartPolicyNever,
+					Containers: []corev1.Container{
+						{
+							Name:            "collector",
+							Image:           *collectorImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									"cpu":    resource.MustParse("7m"),
+									"memory": resource.MustParse("25Mi"),
+								},
+								Limits: corev1.ResourceList{
+									"cpu":    resource.MustParse("8m"),
+									"memory": resource.MustParse("30Mi"),
+								},
+							},
+						},
+					},
+					ServiceAccountName: "eraser-controller-manager",
+				},
+			},
+		},
+	}
+
+	err := r.Create(ctx, job)
+	if err != nil {
+		log.Info("Could not create collector ImageJob")
+		return reconcile.Result{}, err
+	}
+
+	log.Info("Successfully created collector ImageJob", "job", job.Name)
+	return reconcile.Result{}, nil
+}
+
+func (r *Reconciler) updateSharedCRD(ctx context.Context, req ctrl.Request, imageCollector *eraserv1alpha1.ImageCollector) (ctrl.Result, error) {
+	imageCollectorList := &eraserv1alpha1.ImageCollectorList{}
+	if err := r.List(ctx, imageCollectorList); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	items := imageCollectorList.Items
+
+	// store images in map to remove duplicates
+	// map with key: sha id, value: name of image
+	idToTagListMap := make(map[string]string)
+
+	for i := range items {
+		if items[i].Name != collectorShared {
+			temp := items[i].Spec.Images
+			for _, img := range temp {
+				idToTagListMap[img.Digest] = img.Name
+			}
+		}
+	}
+
+	var combined []eraserv1alpha1.Image
+
+	for key, value := range idToTagListMap {
+		combined = append(combined, eraserv1alpha1.Image{Digest: key, Name: value})
+	}
+
+	imageCollector.Spec = eraserv1alpha1.ImageCollectorSpec{Images: combined}
+
+	if err := r.Update(ctx, imageCollector); err != nil {
+		log.Info("Could not update imageCollector spec")
+		return reconcile.Result{}, err
+	}
+
+	// delete individual image collector CRs
```

Comment:
I would put this in a separate method. This method is called `updateSharedCRD`, but it also deletes Kubernetes resources. That's pretty surprising behavior!

I'd split it into 2 functions, `updateSharedCR` and `deleteNodeCRs`. Then in the `Reconcile`, call them successively. That way the side-effects of each method are clear to the reader.